web/pdf_page_view.js

```diff
@@ -583,7 +745,12 @@ class PDFPageView {
 
     if (!keepCanvasWrapper && this.#canvasWrapper) {
       this.#canvasWrapper = null;
-      this.#resetCanvas();
+      this._resetCanvas();
+    }
+
+    if (!preserveDetailViewState) {
+      this.detailView?.reset({ keepCanvas: keepCanvasWrapper });
```

Comment:
To ensure that we leave nothing behind, don't we need to ensure that a detailView canvas also gets removed here?
```suggestion
      this.detailView?.reset();
```

Reply:
Ok yeah the current code is wrong. When opening a PDF and switching toggling on/off an annotations editor, the detail canvas gets left behind in the DOM and never cleaned up. So then if you change the visible area, you get a _third_ canvas.

Not passing `keepCanvas: keepCanvasWrapper` is also wrong, because when toggling the annotations editor we want to keep the detail canvas around.

The correct solution is to only set `this.detailView` when _not_ `keepCanvasWrapper` is false, so that if we still have the canvas we still have the class that tracks it (pushing a commit to fix this now)

Reply:
No, because in this case where we toggle the annotation editor we do want to cancel the in-progress rendering and update the state accordingly, but we don't want to remove the canvas until when the new one is done re-rendering.

So we _could_ pass `preserveDetailViewState: true`, but then we'd need to manually call `this.detailView?.reset({ keepCanvas: true })` after calling `this.reset()` in that place.